includes/validation/class-amp-validation-manager.php

```diff
@@ -168,24 +168,25 @@ class AMP_Validation_Manager {
 	protected static $stylesheet_slug;
 
 	/**
-	 * Add the actions.
+	 * Initialize.
 	 *
-	 * @param array $args {
-	 *     Args.
-	 *
-	 *     @type bool $should_locate_sources Whether to locate sources.
-	 * }
 	 * @return void
 	 */
-	public static function init( $args = [] ) {
-		$args = array_merge(
-			[
-				'should_locate_sources' => self::should_validate_response(),
-			],
-			$args
-		);
+	public static function init() {
+		$should_validate_response = self::should_validate_response();
 
-		self::$should_locate_sources = $args['should_locate_sources'];
+		// Short-circuit validation requests that are unauthorized.
+		if ( $should_validate_response instanceof WP_Error ) {
+			wp_send_json(
+				[
+					'code'    => $should_validate_response->get_error_code(),
+					'message' => $should_validate_response->get_error_message(),
+				],
+				400
```

Comment:
I did have it a 401 initially, but I switched it to be a generic 400 later since I had more conditions in this in `should_validate_response`, but now I can flip it back to 401.